workspace/src/app/views/shared/modals/CreateArtefactModal/ArtefactForms/ParameterAutoCompletion.tsx

```diff
@@ -168,13 +171,44 @@ export const ParameterAutoCompletion = ({
 
     const itemValue = (value: StringOrReifiedValue) => (typeof value === "string" ? value : value.value);
 
+    const handleSearch = React.useCallback(
+        (input: string) => {
+            setSearchQuery(input);
+            setFetchMore(true)
+            setLimit(100)
+            return handleAutoCompleteInput(input, autoCompletion);
+        },
+        [limit]
+    );
+
+    const loadMoreResults = async () => {
+        const newLimit = limit + 100;
```

Comment:
A second request is always made even though the first request can already be less than the limit.